src/test/java/itest/UploadRecordingIT.java

```diff
@@ -110,5 +119,84 @@ public void shouldUploadRecordingToGrafana() throws Exception {
                 String.format("Uploaded: %s\nSet: %s\n", RECORDING_NAME, RECORDING_NAME);
 
         MatcherAssert.assertThat(uploadRespFuture.get(), Matchers.equalTo(expectedUploadResponse));
+
+        // Confirm recording appears in Grafana
+        CompletableFuture<String> getRespFuture = new CompletableFuture<>();
+        webClient
+                .get(8080, "localhost", "/list")
+                .send(
+                        ar -> {
+                            if (assertRequestStatus(ar, getRespFuture)) {
+                                MatcherAssert.assertThat(
+                                        ar.result().statusCode(), Matchers.equalTo(200));
+                                MatcherAssert.assertThat(
+                                        ar.result().getHeader(HttpHeaders.CONTENT_TYPE.toString()),
+                                        Matchers.equalTo(HttpMimeType.JSON.mime()));
+                                getRespFuture.complete(ar.result().bodyAsString());
+                            }
+                        });
+
+        MatcherAssert.assertThat(
+                getRespFuture.get(), Matchers.equalTo(String.format("%s\n", RECORDING_NAME)));
```

Comment:
I would prefer to not check for the `\n` character - either do `getRespFuture().get().trim()` to get a "bare" name string, or perhaps `Arrays.asList(getRespFuture().get().split("[\\s]"))` to turn the result into a list of names.